working/333 - shared memory multithreading/proposal.md

```diff
@@ -842,9 +594,7 @@ class Isolate {
   /// Shared isolate contains a copy of the
```

Comment:
Can a shared function access *local* variables? Are they *shared*?
That is, if I have:
```dart
shared void Function() inc = (){};
shared int Function() val = () => 0;
void initInc() {
  int counter = 0;
  inc = () { counter++; };
  val = () => counter;
}
```
then will it compile? Will it run? Will concurrent accesses update the same variable?

(If you *can't* share closures containing local variables, then some things become much harder. Most things.
Allowing *final* local variables might ease that, but you do want shared mutable state, so it's just a question of whether the compiler boxes the variable or the user has to.)

Reply:
In the _shared **native** memory multithreading_ this code will not compile - because function is not a trivially shareable type. This also means that crossing isolation boundaries will still imply deep copying of objects: e.g. posting a task from a full isolate to a shared isolate will imply copying of the task.